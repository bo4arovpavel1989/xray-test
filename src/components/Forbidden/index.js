import React from 'react'
import { withRouter } from 'react-router-dom'

class Forbidden extends React.Component {
  render () {
    return (
      <div>
        <h1>403 Нет доступа</h1>
        <h3>Залогиньтесь для продолжения</h3>
      </div>
    )
  }
}

export default withRouter(Forbidden)
