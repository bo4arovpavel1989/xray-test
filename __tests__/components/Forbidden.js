import React from 'react';
import { MemoryRouter } from 'react-router-dom'
import Forbidden from '../../src/components/Forbidden';
import Enzyme, { mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('Forbidden component', () => {
  it('should render correct', () => {
      const p404 = mount(
        <MemoryRouter initialEntries={[ { key: 'testKey' } ]}>
          <Forbidden/>
        </MemoryRouter>
      );

      expect(shallowToJson(p404)).toMatchSnapshot()
  })
})
