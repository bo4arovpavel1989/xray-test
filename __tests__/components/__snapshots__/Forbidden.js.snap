// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forbidden component should render correct 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <withRouter(Forbidden)>
      <Route>
        <Forbidden
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "canGo": [Function],
              "createHref": [Function],
              "entries": Array [
                Object {
                  "hash": "",
                  "key": "testKey",
                  "pathname": "/",
                  "search": "",
                },
              ],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "index": 0,
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "key": "testKey",
                "pathname": "/",
                "search": "",
              },
              "push": [Function],
              "replace": [Function],
            }
          }
          location={
            Object {
              "hash": "",
              "key": "testKey",
              "pathname": "/",
              "search": "",
            }
          }
          match={
            Object {
              "isExact": true,
              "params": Object {},
              "path": "/",
              "url": "/",
            }
          }
        >
          <div>
            <h1>
              403 Нет доступа
            </h1>
            <h3>
              Залогиньтесь для продолжения
            </h3>
          </div>
        </Forbidden>
      </Route>
    </withRouter(Forbidden)>
  </Router>
</MemoryRouter>
`;
